src/errors.rs

```diff
@@ -94,3 +109,50 @@ impl<T> From<PoisonError<T>> for ConfigurationAccessError {
         ConfigurationAccessError::LockAcquisitionError
     }
 }
+
+#[derive(Debug, Error)]
+pub(crate) enum SegmentEvaluationError {
+    #[error(transparent)]
+    SegmentEvaluationFailed(#[from] SegmentEvaluationErrorKind),
+
+    #[error("Segment ID '{0}' not found")]
+    SegmentIdNotFound(String),
+}
+#[derive(Debug, Error)]
+#[error("Operation '{}' '{}' '{}' failed to evaluate: {}", segment_rule.attribute_name, segment_rule.operator,  value, source)]
+pub(crate) struct SegmentEvaluationErrorKind {
+    pub(crate) segment: Segment,
+    pub(crate) segment_rule: SegmentRule,
+    pub(crate) value: String,
+    pub(crate) source: CheckOperatorErrorDetail,
+}
```

Comment:
We should be more specific with the data we are using here and store just what we need, otherwise this object is _too big_.

Reply:
yes, I think for the `segment` and `segment_rule` probably the IDs are sufficient. -> follow up PR